### PR TITLE
[INLONG-10091][Sort] Add json format for 1.18

### LINKED ISSUE #10091: [Feature][Sort] Add json format for 1.18
### Description

add support of JSON format for 1.18

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by aloyszhang**
Fixes #10091

### Motivation

Add JSON format for 1.18, and make sure the inlong-sort module can be built with profile v1.18

### Modifications

- add `format-json-v1.18` module 
- resolve flink 1.18 format JSON compatible issue
- make sure the whole project can be build with profile `-Pv1.18`

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: inlong-sort/sort-formats/format-row/format-json-v1.18/src/main/java/org/apache/inlong/sort/formats/json/MysqlBinLogData.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ */
+
+package org.apache.inlong.sort.formats.json;
+
+import org.apache.flink.types.Row;
+
+import java.io.Serializable;
+import java.util.Map;
+
+public class MysqlBinLogData implements Serializable {
```
Comment (baomingyu) [inline inlong-sort/sort-formats/format-row/format-json-v1.18/src/main/java/org/apache/inlong/sort/formats/json/MysqlBinLogData.java:25]: suggest use @Data and @Builder annotation

**Comment by EMsnap** [comment]
I guess module sort-format-json-v1.18 is not built on our build workflow ?

File: inlong-sort/sort-formats/format-row/format-json-v1.18/src/main/java/org/apache/inlong/sort/formats/json/MysqlBinLogData.java (hunk 1 above)
Comment (aloyszhang) [reply inlong-sort/sort-formats/format-row/format-json-v1.18/src/main/java/org/apache/inlong/sort/formats/json/MysqlBinLogData.java:29]: done

**Comment by EMsnap** [comment]
> > I guess module sort-format-json-v1.18 is not built on our build workflow ?
> 
> It's built only by specifying profile v1.18 `mvn clean package -DskipTests -Pv1.18`

Yes and my point is that the workflow hasn't run -Pv1.18 so the module is not build for testifing
image

**Comment by aloyszhang** [comment]
> > > I guess module sort-format-json-v1.18 is not built on our build workflow ?
> > 
> > 
> > It's built only by specifying profile v1.18 `mvn clean package -DskipTests -Pv1.18`
> 
> Yes and my point is that the workflow hasn't run -Pv1.18 so the module is not build for testifing image

You're right, in the current workflow, this module is not built.

**Review by EMsnap — APPROVED** [review]

**Review by baomingyu — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**